extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -124,10 +127,31 @@ private async Task<Stream> CreatePOCOAsync(Container container, CosmosDBInputAtt
                 throw new InvalidOperationException($"Unable to retrieve document with ID '{cosmosAttribute.Id}' and PartitionKey '{cosmosAttribute.PartitionKey}'");
             }
 
-            return item.Content;
+            return JsonSerializer.DeserializeAsync(item.Content, targetType, _serializerOptions)!;
+        }
+
+        private async Task<IList> CreateListAsync(Container container, CosmosDBInputAttribute cosmosAttribute, Type elementType, Type targetType)
+        {
+            var resultType = typeof(List<>).MakeGenericType(elementType);
```

Comment:
Do we intend to support other concrete collection types? Such as `HashSet<T>`? If not the failure is going to be ambiguous - `TryGetCollectionElementType` will pass, but then the cast later on will fail. I suggest we do one of two things:

1. if we don't intend to support the other collection types, we should make that more explicit and fail if the collection type is not explicitly in the set of ones we support.
2. if we do intend to support them, then you would need to branch this logic:
    1. if `targetType` is a concrete type, we can activate it and dynamically call the `.Add` method.
    2. if `targetType` is not concrete, we will provide `List<T>` (I think that covers all collection interfaces we would support? Need to verify that).

Reply:
I don't know if we place to support more collection types any time soon, we can go with option #2 just in case. Or just assume #1 for now and refactor for #2 when we get to that work item?

Reply:
I think what I currently have implemented is #2; if that doesn't look right, I can change this to be option #1 and leave it there for now